### ISSUE TITLE
Ecosystem compatibility with numpy 2.0

### ISSUE THREAD

**Issue by rgommers** [numpy/numpy#26191]
This list tracks the compatibility status of packages that depend on or support NumPy. If "compatible release on PyPI" does not say "yes" but a version number is listed: this is based on plans announced in a tracking issue or other communication by the authors of the package.

_Maintainers: please feel free to edit directly. Others who want to update things: please do comment, or feel free to ping me elsewhere._

| Package name  | Compatible release on PyPI? | Min compatible version | Notes                                                                                                   |
|---------------|-----------------------------|------------------------|---------------------------------------------------------------------------------------------------------|
| Cython        | yes                         | 3.0.4                  | Version is an estimate, it's worked fine for quite a while                                              |
| Pybind11      | yes                         | 2.12.0                 |                                                                                                         |
| Pythran       | yes                         | 0.15.0                 | Version is an estimate, it's worked fine for quite a while                                              |
| SciPy         |                             | 0.13.0                 |                                                                                                         |
| Pandas        |                             |                        |                                                                                                         |
| PyArrow       |                             |                        | https://github.com/apache/arrow/issues/39532                                                            |
| Matplotlib    |                             |                        | https://github.com/matplotlib/matplotlib/issues/26778                                                   |
| scikit-learn  |                             |                        | https://github.com/scikit-learn/scikit-learn/issues/27075                                               |
| scikit-image  |                             |                        | https://github.com/scikit-image/scikit-image/issues/7282                                                |
| statsmodels   |                             |                        |                                                                                                         |
| PyTorch       |                             | 2.3.0                  | https://github.com/pytorch/pytorch/issues/107302                                                        |
| JAX           |                             |                        | https://github.com/google/jax/issues/19246                                                              |
| TensorFlow    |                             |                        |                                                                                                         |
| PyWavelets    | yes                         | 1.6.0                  |                                                                                                         |
| AstroPy       |                             | 6.1.0                  | https://github.com/astropy/astropy/issues/16200                                                         |
| Dask          |                             |                        |                                                                                                         |
| CuPy          |                             | 1.13.1                 | Mostly compatible, some differences like promotion rules (see https://github.com/cupy/cupy/issues/7630) |
| NetworkX      |                             |                        |                                                                                                         |
| Keras         |                             |                        |                                                                                                         |
| Numba         |                             | 0.60                   | https://numba.discourse.group/t/communicating-numpy-2-0-changes-to-numba-users/2457/1                   |
| PyData Sparse |                             |                        |                                                                                                         |
| Xarray        |                             |                        |                                                                                                         |
| AwkwardArray  |                             |                        |                                                                                                         |
| Seaborn       |                             |                        |                                                                                                         |
| SymPy         |                             | 1.12.1                 |                                                                                                         |
| PyMC3         |                             |                        |                                                                                                         |
| BioPython     |                             |                        |                                                                                                         |
| scikit-bio    |                             |                        | https://github.com/scikit-bio/scikit-bio/issues/1964                                                    |
| Shapely       |                             |                        | https://github.com/shapely/shapely/issues/1972                                                          |
| GeoPandas     |                             |                        |                                                                                                         |
|               |                             |                        |                                                                                                         |
|               |                             |                        |                                                                                                         |
|               |                             |                        |                                                                                                         |

**Comment by jakevdp** [comment]
Thanks - for JAX we are planning an 0.4.26 release in the next day or two which will be built against NumPy 2.0.0rc1.

**Comment by jakevdp** [comment]
Also, in case you want to add it, we just released `ml_dtypes` v0.4.0 (https://pypi.org/project/ml-dtypes/) which is compatible with NumPy 2.0.

**Comment by jakirkham** [comment]
Thanks for putting this together Ralf! 🙏

This is incredibly helpful 🙂

**Comment by rgommers** [comment]
Thanks @jakevdp, I added the info for both JAX and `ml_dtypes`.

**Comment by neutrinoceros** [comment]
`unyt` 3.0.2 was also released a couple days ago for compat with numpy 2, in case you'd like to include it !

**Comment by neutrinoceros** [comment]
Also, here's where to track progress for cartopy : https://github.com/SciTools/cartopy/issues/2339

**Comment by rgommers** [comment]
Thanks @neutrinoceros, much appreciated. Added both packages.

**Comment by ianthomas23** [comment]
ContourPy 1.2.1 has just been released on PyPI (https://pypi.org/project/contourpy/1.2.1/) with NumPy 2 compatibility, most relevant PR is contourpy/contourpy#371. It is a compulsory dependency of Matplotlib.

**Comment by jorisvandenbossche** [comment]
For PyArrow, it will be the upcoming 16.0 release that will be the first numpy-2.0-compatible release, but this will only be expected in around 3 weeks at the earliest (but normally certainly before the end of the month). 
For people that need a PyArrow installed in an environment with numpy 2.0, it's worth noting that there are [nightly wheels](https://arrow.apache.org/docs/dev/developers/python.html#installing-nightly-packages) available though that already work with 2.0.

**Comment by rgommers** [comment]
Thanks @ianthomas23, @jorisvandenbossche, @dkbarn - all info added to the table.

**Comment by hawkinsp** [comment]
JAX released 0.4.26 on pypi, which is compatible with NumPy 2.0.

(NumPy folks: congratulations on your imminent v2 release!)

**Comment by ksunden** [comment]
mpl 3.8.4 is out, built with np 2

**Comment by neutrinoceros** [comment]
`h5py` 3.11 is here https://docs.h5py.org/en/stable/whatsnew/3.11.html#new-features

**Comment by rgommers** [comment]
Great, added. Thanks for all the help @neutrinoceros!

**Comment by greglucas** [comment]
Cartopy 0.23 has been released on PyPI and is compatible with numpy 2.0.
https://github.com/SciTools/cartopy/releases/tag/v0.23.0
https://pypi.org/project/Cartopy/0.23.0/

**Comment by neutrinoceros** [comment]
pandas 2.2.2 is out !

**Comment by rgommers** [comment]
As is scikit-learn 1.4.2

**Comment by neutrinoceros** [comment]
I opened a PR for netcdf4-python too https://github.com/Unidata/netcdf4-python/pull/1317

**Comment by neutrinoceros** [comment]
I *think* xarray's progress on this is being discussed in https://github.com/pydata/xarray/issues/8844

**Comment by bashtage** [comment]
statsmodels 0.14.2 is built using NumPy

**Comment by jorisvandenbossche** [comment]
Shapely 2.0.4 is also out with numpy 2.0 compatible wheels

**Comment by rgommers** [comment]
Thanks @bashtage and @jorisvandenbossche, table updated. We're slowly getting there:)

**Comment by bashtage** [comment]
Any insight on numoy 2 builds of conda recipes for conda forge packages?

**Comment by neutrinoceros** [comment]
@bashtage see https://github.com/conda-forge/conda-forge.github.io/issues/1997

**Comment by jakirkham** [comment]
With the table above, it notes "CuPy" as needed "1.13.1". AFAIK that is not a CuPy version. Was that meant to be something else?

cc @leofang (in case you have insight here)

**Comment by rgommers** [comment]
@jakirkham if it doesn't say "yes" in the "compatible release on PyPI" column then it means that the next release is planned to include support. I believe I derived 1.13.1 from comments on the linked issue. But there will be gaps still - it may be 14.0.0 that's the first compatible-enough version. CuPy's plans are still being worked out.

**Comment by jakirkham** [comment]
Ok was it meant to be 13.1.0? Or something else?

**Comment by rgommers** [comment]
Ah yes:) Didn't see my own typo there, fixing now.

**Comment by leofang** [comment]
> But there will be gaps still - it may be 14.0.0 that's the first compatible-enough version. CuPy's plans are still being worked out.

Yes that's right. CuPy has a v14 umbrella tracker noting NumPy 2.0 support: https://github.com/cupy/cupy/issues/8269 but still in planning phase

**Comment by rgommers** [comment]
Today I added info on the new PyArrow and PyTorch releases that support NumPy 2.0. So we're getting somewhere here:)

**Comment by jakirkham** [comment]
@jakevdp do you know what TensorFlow's NumPy 2 plans are? Or who we should ask?

**Comment by jakirkham** [comment]
It looks like Fiona 1.10 will be NumPy 2 compatible based on issue: https://github.com/Toblerity/Fiona/issues/1332

cc @sgillies (to fact check that statement)

**Comment by jakirkham** [comment]
@rgommers could you please add `imageio` to the table with a link to issue ( https://github.com/imageio/imageio/issues/1077 )?

**Comment by sgillies** [comment]
@jakirkham thanks for checking! The fiona package accidentally added a dependency on numpy in one release, but it's been removed. It can be removed from the table ☝️

**Comment by ianthomas23** [comment]
@jakirkham wrote:

> @rgommers could you please add...
> 
> * ContourPy: [[FEATURE] NumPy 2.0 support contourpy/contourpy#378](https://github.com/contourpy/contourpy/issues/378)

This is not necessary, it is already in the list. There was no need to create an issue in ContourPy for this, I did all that was necessary 3 weeks ago.

Perhaps the list should be sorted in alphabetical order?

**Comment by rgommers** [comment]
Thanks everyone for the input, and @jakirkham for following up with a large set of projects.

> Perhaps the list should be sorted in alphabetical order?

I may reorganize it if it gets more unwieldy. Not purely alphabetically though; the top of the table is on purpose for packages that are lowest in the stack, because they were blocking pretty much everything else. So I'll probably want to keep such a set as "must haves, blocking for a final 2.0 release" and then the rest alphabetical.

**Comment by rgommers** [comment]
> It would be really interesting to see the list plotted as a dependency graph, perhaps with individual project bus factor

Any work on this would be great to see indeed. Using tools like https://deps.dev/ or https://libraries.io/ for this to obtain reverse dependency data and then finding ways to analyze/visualize would be interesting.

> shouldn't ContourPy, as a compulsory dependency of Matplotlib, go above it in the list?

Probably not I'd say, since few projects have a direct dependency on it. "Matplotlib works" is pretty much what folks need to know. And that can itself have multiple dependencies. Here is a list: https://deps.dev/pypi/matplotlib/3.8.4/dependencies. I'd prefer to stay mostly out of the business of trying to figure out which of those 10 dependencies are actually blocking/critical.

Of course, the list probably reflects some of my knowledge/biases, it can never be objectively correct or complete. But that's not a real goal here - this is a tool for figuring out where we are in the migration and when we should be able to release 2.0 without creating major issues (or _more_ major issues than necessary) for our users.

**Comment by jakevdp** [comment]
> @jakevdp do you know what TensorFlow's NumPy 2 plans are? Or who we should ask?

I'm not sure what the TensorFlow release plans are. It looks like tensorflow currently [pins numpy<2.0](https://github.com/tensorflow/tensorflow/blob/612cea93bd2f3c6e02ab039dfba337fc74a8fcba/tensorflow/tools/pip_package/setup.py#L88-L89), so it will at least play nicely with numpy 2.0 post-release. @michaelhudgins may be able to say more.

**Comment by MichaelHudgins** [comment]
> > @jakevdp do you know what TensorFlow's NumPy 2 plans are? Or who we should ask?
> 
> I'm not sure what the TensorFlow release plans are. It looks like tensorflow currently [pins numpy<2.0](https://github.com/tensorflow/tensorflow/blob/612cea93bd2f3c6e02ab039dfba337fc74a8fcba/tensorflow/tools/pip_package/setup.py#L88-L89), so it will at least play nicely with numpy 2.0 post-release. @MichaelHudgins may be able to say more.

Thanks for the ping Jake.

TensorFlow is targeting a branch cut end of May, we are unsure if numpy 2.0 support will make it but we are going to try and i should have better insight next week if the update will make next release.  Like Jake said TensorFlow does have a <2.0 pin in at least both 2.16.1 and 2.15.1 which are our two supported versions currently, so I don't think we should immediately break if TensorFlow does not get the update done before a 2.0 release.

**Comment by rootsmusic** [comment]
What about [dpnp](https://intelpython.github.io/dpnp/overview.html)?

**Comment by jakirkham** [comment]
> What about [dpnp](https://intelpython.github.io/dpnp/overview.html)?

@rootsmusic would suggest [checking their repo]( https://github.com/IntelPython/dpnp ) and if it is unclear raising an issue on their repo asking their plans

**Comment by bryevdv** [comment]
@jakirkham AFAICT Existing Bokeh 3.4.1 already works with Numpy 2.0. There is only one failing unit test for me locally and that only due to the text of an expected exception not showing `bool_` as it did before. Manual spot-testing seems to work fine as well. Unfortunately, I have not been able to successfully create any CI environments. That is because, while a PyPI-installed Pandas 2.2.2 works fine, the CI envs install pandas 2.2.2 from conda-forge and that does *not* work (a situation I can reproduce locally, see details). If you have any advice about the condaf-forge pandas, please advise. 

<details>

```
~/work/bokeh/conda bv/numpy-2.0
dev312 ❯ ipython
Python 3.12.0 | packaged by conda-forge | (main, Oct  3 2023, 08:36:57) [Clang 15.0.7 ]
Type 'copyright', 'credits' or 'license' for more information
IPython 8.18.1 -- An enhanced Interactive Python. Type '?' for help.

In [1]: import pandas as pd

In [2]:
Do you really want to exit ([y]/n)?

~/work/bokeh/conda bv/numpy-2.0 11s
dev312 ❯ pip uninstall pandas
Found existing installation: pandas 2.2.2
Uninstalling pandas-2.2.2:
  Would remove:
    /Users/bryan/anaconda3/envs/dev312/lib/python3.12/site-packages/pandas-2.2.2.dist-info/*
    /Users/bryan/anaconda3/envs/dev312/lib/python3.12/site-packages/pandas/*
Proceed (Y/n)?
  Successfully uninstalled pandas-2.2.2

~/work/bokeh/conda bv/numpy-2.0 11s
dev312 ❯ conda list pandas
# packages in environment at /Users/bryan/anaconda3/envs/dev312:
#
# Name                    Version                   Build  Channel
pandas-datareader         0.10.0             pyh6c4a22f_0    conda-forge
pandas-stubs              2.1.4.231227             pypi_0    pypi

~/work/bokeh/conda bv/numpy-2.0
dev312 ❯ conda install -c conda-forge  "pandas==2.2.2"
Channels:
 - conda-forge
 - defaults
Platform: osx-arm64
Collecting package metadata (repodata.json): done
Solving environment: done

## Package Plan ##

  environment location: /Users/bryan/anaconda3/envs/dev312

  added / updated specs:
    - pandas==2.2.2


The following NEW packages will be INSTALLED:

  pandas             conda-forge/osx-arm64::pandas-2.2.2-py312h88edd18_0
  python-dateutil    conda-forge/noarch::python-dateutil-2.9.0-pyhd8ed1ab_0
  python-tzdata      conda-forge/noarch::python-tzdata-2024.1-pyhd8ed1ab_0
  pytz               conda-forge/noarch::pytz-2024.1-pyhd8ed1ab_0


Proceed ([y]/n)?


Downloading and Extracting Packages:

Preparing transaction: done
Verifying transaction: done
Executing transaction: done

~/work/bokeh/conda bv/numpy-2.0 16s
dev312 ❯ ipython
Python 3.12.0 | packaged by conda-forge | (main, Oct  3 2023, 08:36:57) [Clang 15.0.7 ]
Type 'copyright', 'credits' or 'license' for more information
IPython 8.18.1 -- An enhanced Interactive Python. Type '?' for help.

In [1]: import pandas as pd

A module that was compiled using NumPy 1.x cannot be run in
NumPy 2.0.0rc1 as it may crash. To support both 1.x and 2.x
versions of NumPy, modules must be compiled with NumPy 2.0.
Some module may need to rebuild instead e.g. with 'pybind11>=2.12'.

If you are a user of the module, the easiest solution will be to
downgrade to 'numpy<2' or try to upgrade the affected module.
We expect that some modules will need time to support NumPy 2.

Traceback (most recent call last):  File "/Users/bryan/anaconda3/envs/dev312/bin/ipython", line 10, in <module>
    sys.exit(start_ipython())
  File "/Users/bryan/anaconda3/envs/dev312/lib/python3.12/site-packages/IPython/__init__.py", line 129, in start_ipython
    return launch_new_instance(argv=argv, **kwargs)
  File "/Users/bryan/anaconda3/envs/dev312/lib/python3.12/site-packages/traitlets/config/application.py", line 1077, in launch_instance
    app.start()
  File "/Users/bryan/anaconda3/envs/dev312/lib/python3.12/site-packages/IPython/terminal/ipapp.py", line 317, in start
    self.shell.mainloop()
  File "/Users/bryan/anaconda3/envs/dev312/lib/python3.12/site-packages/IPython/terminal/interactiveshell.py", line 887, in mainloop
    self.interact()
  File "/Users/bryan/anaconda3/envs/dev312/lib/python3.12/site-packages/IPython/terminal/interactiveshell.py", line 880, in interact
    self.run_cell(code, store_history=True)
  File "/Users/bryan/anaconda3/envs/dev312/lib/python3.12/site-packages/IPython/core/interactiveshell.py", line 3048, in run_cell
    result = self._run_cell(
  File "/Users/bryan/anaconda3/envs/dev312/lib/python3.12/site-packages/IPython/core/interactiveshell.py", line 3103, in _run_cell
    result = runner(coro)
  File "/Users/bryan/anaconda3/envs/dev312/lib/python3.12/site-packages/IPython/core/async_helpers.py", line 129, in _pseudo_sync_runner
    coro.send(None)
  File "/Users/bryan/anaconda3/envs/dev312/lib/python3.12/site-packages/IPython/core/interactiveshell.py", line 3308, in run_cell_async
    has_raised = await self.run_ast_nodes(code_ast.body, cell_name,
  File "/Users/bryan/anaconda3/envs/dev312/lib/python3.12/site-packages/IPython/core/interactiveshell.py", line 3490, in run_ast_nodes
    if await self.run_code(code, result, async_=asy):
  File "/Users/bryan/anaconda3/envs/dev312/lib/python3.12/site-packages/IPython/core/interactiveshell.py", line 3550, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "<ipython-input-1-7dd3504c366f>", line 1, in <module>
    import pandas as pd
  File "/Users/bryan/anaconda3/envs/dev312/lib/python3.12/site-packages/pandas/__init__.py", line 49, in <module>
    from pandas.core.api import (
  File "/Users/bryan/anaconda3/envs/dev312/lib/python3.12/site-packages/pandas/core/api.py", line 1, in <module>
    from pandas._libs import (
  File "/Users/bryan/anaconda3/envs/dev312/lib/python3.12/site-packages/pandas/_libs/__init__.py", line 17, in <module>
    import pandas._libs.pandas_datetime  # noqa: F401 # isort: skip # type: ignore[reportUnusedImport]
---------------------------------------------------------------------------
ImportError                               Traceback (most recent call last)
File ~/anaconda3/envs/dev312/lib/python3.12/site-packages/numpy/core/_multiarray_umath.py:44, in __getattr__(attr_name)
     39     # Also print the message (with traceback).  This is because old versions
     40     # of NumPy unfortunately set up the import to replace (and hide) the
     41     # error.  The traceback shouldn't be needed, but e.g. pytest plugins
     42     # seem to swallow it and we should be failing anyway...
     43     sys.stderr.write(msg + tb_msg)
---> 44     raise ImportError(msg)
     46 ret = getattr(_multiarray_umath, attr_name, None)
     47 if ret is None:

ImportError:
A module that was compiled using NumPy 1.x cannot be run in
NumPy 2.0.0rc1 as it may crash. To support both 1.x and 2.x
versions of NumPy, modules must be compiled with NumPy 2.0.
Some module may need to rebuild instead e.g. with 'pybind11>=2.12'.

If you are a user of the module, the easiest solution will be to
downgrade to 'numpy<2' or try to upgrade the affected module.
We expect that some modules will need time to support NumPy 2.


---------------------------------------------------------------------------
ImportError                               Traceback (most recent call last)
Cell In[1], line 1
----> 1 import pandas as pd

File ~/anaconda3/envs/dev312/lib/python3.12/site-packages/pandas/__init__.py:49
     46 # let init-time option registration happen
     47 import pandas.core.config_init  # pyright: ignore[reportUnusedImport] # noqa: F401
---> 49 from pandas.core.api import (
     50     # dtype
     51     ArrowDtype,
     52     Int8Dtype,
     53     Int16Dtype,
     54     Int32Dtype,
     55     Int64Dtype,
     56     UInt8Dtype,
     57     UInt16Dtype,
     58     UInt32Dtype,
     59     UInt64Dtype,
     60     Float32Dtype,
     61     Float64Dtype,
     62     CategoricalDtype,
     63     PeriodDtype,
     64     IntervalDtype,
     65     DatetimeTZDtype,
     66     StringDtype,
     67     BooleanDtype,
     68     # missing
     69     NA,
     70     isna,
     71     isnull,
     72     notna,
     73     notnull,
     74     # indexes
     75     Index,
     76     CategoricalIndex,
     77     RangeIndex,
     78     MultiIndex,
     79     IntervalIndex,
     80     TimedeltaIndex,
     81     DatetimeIndex,
     82     PeriodIndex,
     83     IndexSlice,
     84     # tseries
     85     NaT,
     86     Period,
     87     period_range,
     88     Timedelta,
     89     timedelta_range,
     90     Timestamp,
     91     date_range,
     92     bdate_range,
     93     Interval,
     94     interval_range,
     95     DateOffset,
     96     # conversion
     97     to_numeric,
     98     to_datetime,
     99     to_timedelta,
    100     # misc
    101     Flags,
    102     Grouper,
    103     factorize,
    104     unique,
    105     value_counts,
    106     NamedAgg,
    107     array,
    108     Categorical,
    109     set_eng_float_format,
    110     Series,
    111     DataFrame,
    112 )
    114 from pandas.core.dtypes.dtypes import SparseDtype
    116 from pandas.tseries.api import infer_freq

File ~/anaconda3/envs/dev312/lib/python3.12/site-packages/pandas/core/api.py:1
----> 1 from pandas._libs import (
      2     NaT,
      3     Period,
      4     Timedelta,
      5     Timestamp,
      6 )
      7 from pandas._libs.missing import NA
      9 from pandas.core.dtypes.dtypes import (
     10     ArrowDtype,
     11     CategoricalDtype,
   (...)
     14     PeriodDtype,
     15 )

File ~/anaconda3/envs/dev312/lib/python3.12/site-packages/pandas/_libs/__init__.py:17
     13 # Below imports needs to happen first to ensure pandas top level
     14 # module gets monkeypatched with the pandas_datetime_CAPI
     15 # see pandas_datetime_exec in pd_datetime.c
     16 import pandas._libs.pandas_parser  # isort: skip # type: ignore[reportUnusedImport]
---> 17 import pandas._libs.pandas_datetime  # noqa: F401 # isort: skip # type: ignore[reportUnusedImport]
     18 from pandas._libs.interval import Interval
     19 from pandas._libs.tslibs import (
     20     NaT,
     21     NaTType,
   (...)
     26     iNaT,
     27 )

ImportError: numpy.core.multiarray failed to import

In [2]:
Do you really want to exit ([y]/n)?

~/work/bokeh/conda bv/numpy-2.0 4m 41s
dev312 ❯ conda list pandas
# packages in environment at /Users/bryan/anaconda3/envs/dev312:
#
# Name                    Version                   Build  Channel
pandas                    2.2.2           py312h88edd18_0    conda-forge
pandas-datareader         0.10.0             pyh6c4a22f_0    conda-forge
pandas-stubs              2.1.4.231227             pypi_0    pypi
``

</details>

**Comment by bashtage** [comment]
@bryevdv conda-froge hasn't fully worked out how to build with NumPy 2 yet. conda packages built for NumPy 2-compatible projects are still compiled against the default NumPy version and so will show that warning when NumPy 2 is installed in the environment.

**Comment by keewis** [comment]
`hypothesis` is compatible with numpy 2 since `6.100.2` (the tracking issue was HypothesisWorks/hypothesis#3950).

**Comment by h-vetinari** [comment]
> conda-forge hasn't fully worked out how to build with NumPy 2 yet.

Actually we have it figured out. We just need to make a choice about how to roll it out, which has a couple of choices with different trade-offs, and none perfect.

**Comment by jorisvandenbossche** [comment]
@rgommers you can update the GeoPandas row as being released (https://github.com/geopandas/geopandas/releases/tag/v0.14.4)

**Comment by neutrinoceros** [comment]
astropy 6.1.0 is out !

**Comment by h-vetinari** [comment]
boost/python sits pretty deep in the stack from the POV of the conda-forge dependency graph (though admittedly the graph conflates packages depending on numpy & libboost with those depending on numpy & libboost-python, because all boost components are build in the same feedstock). So from that POV, it becomes quite important to solve https://github.com/boostorg/python/issues/431.

**Comment by jakirkham** [comment]
IIUC Boost.Python fixed this with PR: https://github.com/boostorg/python/pull/432

 Have we tried that patch?

**Comment by h-vetinari** [comment]
Yes, Matti pointed out that patch to me (it wasn't linked in the issue so I didn't see it), and it works fine when backported to 1.84 and even 1.82.

**Comment by jakirkham** [comment]
@rgommers could we please update these entries in the table in the OP?


Package name | Compatible release on PyPI? | Min compatible version | Notes
-- | -- | -- | --
numcodecs |  yes  | 0.12.1 (likely older versions too; has been stable for a while)| zarr-developers/numcodecs#521
Zarr |  yes  |  2.18.0  | zarr-developers/zarr-python#1818

**Comment by rgommers** [comment]
Updated for all recent comments, thanks all!

**Comment by jakirkham** [comment]
Thanks Ralf! 🙏

Looks like nearly all libraries have some kind of issue/PR reference (or already a working release)

One that appears to be missed was Keras, so have raised upstream issue: https://github.com/keras-team/keras/issues/19691

**Comment by jakirkham** [comment]
> TensorFlow is targeting a branch cut end of May, we are unsure if numpy 2.0 support will make it but we are going to try and i should have better insight next week if the update will make next release. Like Jake said TensorFlow does have a <2.0 pin in at least both 2.16.1 and 2.15.1 which are our two supported versions currently, so I don't think we should immediately break if TensorFlow does not get the update done before a 2.0 release.

Thanks @MichaelHudgins for the insight here

Am curious how things are looking for TensorFlow + NumPy 2. Were you able to get more insight in the past 2 weeks?

**Comment by h-vetinari** [comment]
The boost version in the table is wrong. https://github.com/boostorg/python/commit/0474de0f6cc9c6e7230aeb7164af2f7e4ccf74bf is not part of 1.85, but will be in 1.86.

PS. Sidenote: it would be helpful if the rest of the table (perhaps excluding some "most important packages") were sorted alphabetically, as it's getting a bit unwieldy to find specific packages.

**Comment by bryevdv** [comment]
> it would be helpful if the rest of the table (perhaps excluding some "most important packages") were sorted alphabetically, as it's getting a bit unwieldy to find specific packages.

Agreed, if "important" packages need to be specially identified, they can be bolded or a column with an X or check mark or whatever

**Comment by bersbersbers** [comment]
I could not find an issue in their [bug tracker](https://bugreports.qt.io/browse/), but some testing implies that `PySide6` works from version 6.7.0 on:

```
# exits with "A module that was compiled using NumPy 1.x cannot be run in NumPy 2.0.0rc2 as it may crash."
uv pip install numpy==2.0.0rc2 PySide6-Essentials==6.6.3.1 && python -c "from PySide6 import QtWidgets"

# works fine
uv pip install numpy==2.0.0rc2 PySide6-Essentials==6.7.0   && python -c "from PySide6 import QtWidgets"
```

**Comment by jakirkham** [comment]
@rgommers have a few updates batched below for the table

GDAL 3.9.0 is released (with NumPy 2 support)! 🎉
* https://lists.osgeo.org/pipermail/gdal-dev/2024-May/058988.html

Also can we please add these NumPy 2 tracking issues to the table:

* TensorFlow: https://github.com/tensorflow/tensorflow/issues/67291
* Keras: https://github.com/keras-team/keras/issues/19691

At this point Numba's 0.60.0rc1 is packaged ( https://github.com/numba/numba/issues/9544#issuecomment-2113293982 ). Working on a conda-forge RC package for it as well. Would start pointing libraries downstream of Numba to these

Boost.Python should be fixed in the 1.86.0 release (we narrowly missed 1.85.0)

Dask is looking promising ( https://github.com/dask/dask/issues/11066#issuecomment-2113284358 ). Though still want to double check things there

**Comment by rgommers** [comment]
Thanks all, updated for all comments.

> PS. Sidenote: it would be helpful if the rest of the table (perhaps excluding some "most important packages") were sorted alphabetically, as it's getting a bit unwieldy to find specific packages.

Okay, I reordered it. The initial order has served its purpose, everything was blocked on a small set of packages at the start but those all did their release.

**Comment by hameerabbasi** [comment]
> Lastly opened an issue for Sparse: [pydata/sparse#680](https://github.com/pydata/sparse/issues/680)

`sparse` release is up on PyPI: https://pypi.org/project/sparse/0.15.2/

**Comment by jakirkham** [comment]
Dask 2024.5.1 works with NumPy 2

ref: https://docs.dask.org/en/stable/changelog.html#numpy-2-0-support

**Comment by rgommers** [comment]
Great, thanks @hameerabbasi and @jakirkham. Updated the table for those, and also added RAPIDS (linking to your tracking issue @jakirkham).

We're getting there - we've had ~7 weeks since RC1 and are about 75% through the list in this issue. In the community meeting next Wednesday we should probably discuss picking a release date in June.

**Comment by bnavigator** [comment]
What about (optional) inter-dependencies? Pandas and numexpr both claim to be compatible, but together they choke on the changed numpy 2 type repr of scalars: https://github.com/pandas-dev/pandas/issues/58548

**Comment by rgommers** [comment]
I'd see that as a regular bug - a few test failures in a combo of two packages that were not tested together in CI is common, but doesn't detract (from the purposes of this tracking issue at least) from them both being NumPy 2.0-compatible.

**Comment by rgommers** [comment]
Posting here as well for completeness (xref https://github.com/numpy/numpy/issues/24300#issuecomment-2125643448): **release date will be June 16th**.

**Comment by peterjc** [comment]
For Biopython we do have C code using NumPy, but my understanding is we are insulated from the C API changes due to using the buffer protocol rather than the NumPy C API.

I expect a "fully compatible" Biopython 1.84 in early June, but all this addresses is the scalar repr changes to work on recent Numpy v1 and the v2 release candidates https://github.com/biopython/biopython/issues/4676

https://lists.open-bio.org/pipermail/biopython/2024-May/017049.html

**Comment by jakirkham** [comment]
> Great, thanks @hameerabbasi and @jakirkham. Updated the table for those, and also added RAPIDS (linking to your tracking issue @jakirkham).

Thanks Ralf! 🙏 Also thank you for including us 🙂

**Comment by hcho3** [comment]
Treelite 4.2.1 is compatible with NumPy 2.0

**Comment by oscarbenjamin** [comment]
I just pushed SymPy 1.12.1 to PyPI which is compatible with NumPy 2.0.

**Comment by rgommers** [comment]
Thanks @hcho3 and @oscarbenjamin! We're getting there:)

**Comment by jakirkham** [comment]
Given we haven't heard anything on the TensorFlow side, maybe we should drop the "(maybe in May)" note above (as we are nearly in June 😉)

**Comment by rgommers** [comment]
Sure, dropped "maybe in May". Of all the tracked packages, TensorFlow is the one I'm least worried about, because they have upper bounds in place and it's common for TF to be behind the latest `numpy`.

**Comment by MichaelHudgins** [comment]
> Given we haven't heard anything on the TensorFlow side, maybe we should drop the "(maybe in May)" note above (as we are nearly in June 😉)

Sorry for the late reply, TF is working on the upgrade but it may not make the upcoming branchcut to ensure all cases around  datatype promotion are handled correctly.  Either way it should land in our nightly in the next few weeks, but as mentioned TF has an upperbound in place for our supported 2 previous versions.

**Comment by rgommers** [comment]
Thanks for the update @MichaelHudgins, that sounds great.

**Comment by jakirkham** [comment]
Thanks Michael! 🙏

**Comment by jakirkham** [comment]
Could we please add these issues to the table?

| Package name | Compatible release on PyPI? | Min compatible version | Notes |
| -- | -- | -- | -- |
| LightGBM                               |    | 4.4.0 (planned)                | https://github.com/microsoft/LightGBM/issues/6454 & https://github.com/microsoft/LightGBM/pull/6439                                                                                                    |
| mlpack                                 |    |                                | https://github.com/mlpack/mlpack/issues/3723                                                                                                                                                           |
| pvlib                                  | Y  | 0.10.5                         | https://github.com/pvlib/pvlib-python/issues/2026                                                                                                                                                      |
| Py-ART / xarm-pyart                    | Y  | 1.18.2                         | https://github.com/ARM-DOE/pyart/issues/1550                                                                                                                                                           |
| QuTiP                                  |    |                                | https://github.com/qutip/qutip/pull/2421                                                                                                                                                               |

**Comment by conradsnicta** [comment]
Please add CARMA to the table:
* git repo: https://github.com/RUrlus/carma
* NumPy 2.0 support related issue: https://github.com/RUrlus/carma/issues/129

**Comment by rgommers** [comment]
Thanks @jakirkham and @conradsnicta, all done.

**Comment by rgommers** [comment]
@jakirkham I've sent you an invite with triage permissions - please feel free to make edits directly (and thanks again for all the help!).

**Comment by jameslamb** [comment]
Could this one for `catboost` be added? https://github.com/catboost/catboost/issues/2671

That project does not have a compatible release, and based on comments there seems to be planning a `< 2.0` pin for now.

**Comment by rgommers** [comment]
Thanks @jameslamb, added.

**Comment by jakirkham** [comment]
> @jakirkham I've sent you an invite with triage permissions - please feel free to make edits directly (and thanks again for all the help!).

Thanks Ralf! 🙏

Accepted 😄

Interestingly I'm not able to edit the OP though

<img width="1840" alt="Screenshot 2024-05-30 at 11 03 29 AM" src="https://github.com/numpy/numpy/assets/3019665/8069be57-f2da-4969-8844-bb686dd2db98">

Guessing we would need edit privileges for that

Maybe we could move this into another repo or HackMD where we could liberally hand out permissions for editing? What do you think?

**Comment by jakirkham** [comment]
Could we please add these to the list as well?

Sorry would do myself, but wasn't able to (as noted above)

| Package name                                                                     | Compatible release on PyPI? | Min compatible version                                       | Notes |
| ------------ | -------------------------------------------------------------------------------- | ---------------------- | ------------------------------------------------------------ |
| adaptive                 |                                                                                                                                                                  |                                              | https://github.com/python-adaptive/adaptive/pull/458                                                                     |
| clawpack                 |                                                                                                                                                                  |                                              | https://github.com/clawpack/clawpack/issues/252                                                                          |
| ITK                      |                                                                                                                                                                  |                                              | https://github.com/InsightSoftwareConsortium/ITK/issues/4700                                                             |
| pint                     |                                                                                                                                                                  |                                              | https://github.com/hgrecco/pint/issues/1974                                                                              |
| PyKrige                  | N? https://github.com/GeoStat-Framework/PyKrige/pull/290#issuecomment-2141034757                                                                                 | 1.7.2                                        | https://github.com/GeoStat-Framework/PyKrige/pull/290                                                                    |
| scipp                    |                                                                                                                                                                  |                                              | https://github.com/scipp/scipp/pull/3427                                                                                 |

**Comment by basnijholt** [comment]
I just released [Adaptive](https://github.com/python-adaptive/adaptive) [v1.3.0](https://pypi.org/project/adaptive/1.3.0/) which makes it compatible with NumPy 2.0 :tada:

**Comment by jakirkham** [comment]
Congratulations 🥳  Thanks Bas! 🙏

Have updated the table in my comment above to reflect this. Please let me know if I've missed anything

**Comment by thewtex** [comment]
Thanks for the work on this!

Can we please add:

> ITK	Y	5.4.0	[InsightSoftwareConsortium/ITK#4700](https://github.com/InsightSoftwareConsortium/ITK/issues/4700)

?

**Comment by cgohlke** [comment]
[Imagecodecs 2024.6.1](https://pypi.org/project/imagecodecs/) supports numpy 2.

**Comment by rgommers** [comment]
Thanks everyone! I made all requested edits.

> Maybe we could move this into another repo or HackMD where we could liberally hand out permissions for editing? What do you think?

I'd rather not move the authoritative version, so close to the finish line. Having an edit history with diffs is also useful. However, easily sync-able edits are useful, so here is an editable HackMD version: https://hackmd.io/B_zU26JfSI-703Xz85Ie_Q?view. Every user who is signed in to HackMD should be able to edit that.

**Comment by NicolasHug** [comment]
Hi all, torchvision 0.19 will be fully compatible with numpy 2.0.
The current release (0.18) is 98% comptible with 2.0, there's only one torchvision transform (`adjust_hue` and its dependents) who would fail on 2.0 due to a uint8 overflow.

**Comment by andrewgsavage** [comment]
pint 0.24 is compatible and released on pypi

**Comment by jakirkham** [comment]
> > Maybe we could move this into another repo or HackMD where we could liberally hand out permissions for editing? What do you think?
> 
> I'd rather not move the authoritative version, so close to the finish line. Having an edit history with diffs is also useful. However, easily sync-able edits are useful, so here is an editable HackMD version: https://hackmd.io/B_zU26JfSI-703Xz85Ie_Q?view. Every user who is signed in to HackMD should be able to edit that.

Completely understand. Got the sense that you wanted help in making (or batching) edits. So was trying to think of options where the workload could be shared more easily 🙂

<hr>

Have included the updates before this comment and after Ralf's last comment in the HackMD. Please review (and feel free to edit if needed)

**Comment by RUrlus** [comment]
> Please add CARMA to the table:
> 
> * git repo: https://github.com/RUrlus/carma
> * NumPy 2.0 support related issue: [NumPy 2.0 support RUrlus/carma#129](https://github.com/RUrlus/carma/issues/129)

CARMA v0.8.0 adds support for NumPy 2.0. There are no PyPi releases (C++ header only) so that is NA.
Props to the NumPy developers (@seberg in particular) for making it a smooth transition!

**Comment by kmaehashi** [comment]
[CuPy v13.2](https://github.com/cupy/cupy/releases/tag/v13.2.0) is just out, including support for NumPy 2.0.

Note that the v13.x series is a stable release stream that uses NumPy v1.26 as a baseline.  Full API compatibility with NumPy v2, including new APIs, is going to be added to the v14.x series under development.

**Comment by keewis** [comment]
I've just released `xarray=2024.06.0`, which is compatible with numpy 2.

**Comment by rgommers** [comment]
Numba 0.60.0 is out too.

**Comment by rootsmusic** [comment]
> HackMD version: https://hackmd.io/B_zU26JfSI-703Xz85Ie_Q?view

@rgommers This issue's table synced with HackMD, because I don't see modin (added in HackMD).

**Comment by jameslamb** [comment]
`lightgbm` 4.4.0, which is compatible with `numpy` 2.0, is now up on PyPI.

LightGBM can get a "yes" in the "compatible release on PyPI" column.

**Comment by rgommers** [comment]
Thanks both - added Modin and LightGBM to the table.

**Comment by charris** [comment]
Pushed this off to 2.0.1 to clear the 2.0.0 milestone.

**Comment by adamjstewart** [comment]
lightly still does not support numpy 2: https://github.com/lightly-ai/lightly/issues/1558

**Comment by neutrinoceros** [comment]
NetCDF4 1.7.0 was released a couple days ago and is compatible.

**Comment by seberg** [comment]
Thanks, updated bottlneck, NetCDF4, and added lightly.

**Comment by rgommers** [comment]
I also updated OpenCV, GNURadio and fitsio - they all did their releases.

**Comment by drammock** [comment]
MNE-Python: 1.7 is numpy2 compatible, and [is on PyPI](https://pypi.org/project/mne/) (and conda-forge)

**Comment by greglandrum** [comment]
@rgommers we just merged the PR for the RDKit to work with numpy2.0. The changes will be in the 2024.03.4 release, which wil be out before the end of the month.

**Comment by henryiii** [comment]
Awkward is above, but wanted to also mention two more in the top [8K on PyPI](https://hugovk.github.io/top-pypi-packages/): boost-histogram 1.4.1 (and therefore Hist) supported NumPy 2 on Mar 30, 2024 and iminuit 2.26.0 on Jun 3, 2024.

**Comment by seberg** [comment]
Thanks Henry, added both.

**Comment by jakirkham** [comment]
`imageio` version `2.34.2` supports NumPy 2

xref: https://github.com/imageio/imageio/issues/1090

**Comment by peterjc** [comment]
Biopython 1.84 is out now and the tests pass under Numpy 2.0.0 (and recent v1 releases).

**Comment by adamjstewart** [comment]
TorchGeo 0.6.0 will be the first release compatible with numpy 2: https://github.com/microsoft/torchgeo/pull/2151

**Comment by mbyt** [comment]
The released polars 1.1.0 has now numpy 2.0 support: https://github.com/pola-rs/polars/releases/tag/py-1.1.0

**Comment by adamjstewart** [comment]
Tensorboard 1.17.0 does not support numpy 2: https://github.com/tensorflow/tensorboard/issues/6869, https://github.com/tensorflow/tensorboard/issues/6874

This has been fixed in master: https://github.com/tensorflow/tensorboard/pull/6871

Unclear if this will be included in 1.17.1 or 1.18.0.

**Comment by jakirkham** [comment]
Think we can now add...

Package name | Compatible release on PyPI? | Min compatible version | Notes
--- | --- | --- | ---
scipp |  Yes  | 24.06.0  | https://github.com/scipp/scipp/pull/3427

**Comment by jakirkham** [comment]
Also

Package name | Compatible release on PyPI? | Min compatible version | Notes
--- | --- | --- | ---
reikna |  Yes  | 0.9.0  | https://github.com/fjarri/reikna/pull/69

**Comment by neutrinoceros** [comment]
Pyvista 0.44.0 was also just released with numpy 2 compat https://github.com/pyvista/pyvista/releases/tag/v0.44.0

**Comment by agriyakhetarpal** [comment]
Pyodide hopes to include NumPy v2 in the next major release and is being tracked in https://github.com/pyodide/pyodide/pull/4925.

**Comment by jakirkham** [comment]
Also

Package name | Compatible release on PyPI? | Min compatible version | Notes
--- | --- | --- | ---
CPNest |  Yes  | 0.11.6  | https://github.com/johnveitch/cpnest/issues/92

**Comment by miikkas** [comment]
PyEDFlib doesn't have a NumPy 2.0 compatible release yet, but here's the tracking issue: https://github.com/holgern/pyedflib/issues/259

**Comment by sammaxwellxyz** [comment]
torchvision 0.19.0 is now on pypi

**Comment by rgommers** [comment]
> torchvision 0.19.0 is now on pypi

Thanks. We need 0.19.1 though - as the comment in the table says, the 0.19.0 Windows wheels do not support NumPy 2.0 (other platforms work fine).

**Comment by stonebig** [comment]
botorch-1.11.3 seems not compatible (Windows issue) https://github.com/pytorch/botorch/releases/tag/v0.11.3

**Comment by zoj613** [comment]
[polyagamma v2.0.0](https://github.com/zoj613/polyagamma) (used as an optional dependency by PyMC) supports numpy 2.0 and is out on [PyPI](https://pypi.org/project/polyagamma/#files)

**Comment by BartSchuurmans** [comment]
Pyomo does not have a compatible release yet, but a PR fixing support has been merged, so it will probably be in the next release: https://github.com/Pyomo/pyomo/pull/3292

**Comment by blnicho** [comment]
> Pyomo does not have a compatible release yet, but a PR fixing support has been merged, so it will probably be in the next release: [Pyomo/pyomo#3292](https://github.com/Pyomo/pyomo/pull/3292)

Pyomo 6.8.0 is compatible with NumPy 2.0 and now on PyPI

**Comment by agriyakhetarpal** [comment]
[`autograd` v1.7.0](https://github.com/HIPS/autograd/releases/tag/v1.7.0) has added support for NumPy 2.0.

**Comment by adamjstewart** [comment]
TorchGeo 0.6.0 was just released on PyPI and includes compatibility with NumPy 2.

**Comment by adamjstewart** [comment]
Not as popular, but for completeness, cmocean 3.1.0 is the first release with numpy 2 support: https://github.com/matplotlib/cmocean/pull/99

**Comment by bashtage** [comment]
A few lates ones;

* linearmodels 6.0.0 (April 16)
* arch 7.0.0 (April 16)
* randomgen 2.0.0 (September 23)

**Comment by jameslamb** [comment]
`sagemaker` is still pinned to `<2.0`: https://github.com/aws/sagemaker-python-sdk/issues/4882

That's the official Python client library for AWS's SageMaker service ([AWS docs](https://docs.aws.amazon.com/sagemaker/latest/dg/api-and-sdk-reference-overview.html)).

It's downloaded from PyPI around 1M times a day (https://pypistats.org/packages/sagemaker), so I think it's popular enough to warrant inclusion here.

**Comment by adamjstewart** [comment]
PESQ does not yet have numpy 2 support: https://github.com/ludlows/PESQ/issues/54

**Comment by mattip** [comment]
@agriyakhetarpal I added those to the table, thanks.

**Comment by vishwin** [comment]
boostorg/python#431 boostorg/python#432 boostorg/python#443 are not in 1.86.0, but are tagged in 1.87.0.beta1.

**Comment by rgommers** [comment]
> [boostorg/python#431](https://github.com/boostorg/python/issues/431) [boostorg/python#432](https://github.com/boostorg/python/pull/432) [boostorg/python#443](https://github.com/boostorg/python/pull/443) are not in 1.86.0, but are tagged in 1.87.0.beta1.

Thanks for pointing that out, fixed in the table now.

**Comment by agriyakhetarpal** [comment]
`river` (https://github.com/online-ml/river, https://pypi.org/project/river/) now builds against NumPy 2.

Package name | Compatible release on PyPI? | Min compatible version | Notes
-- | -- | -- | --
river | yes | 0.22.0 | https://github.com/online-ml/river/pull/1632, https://github.com/online-ml/river/issues/1649

**Comment by poedator** [comment]
please add `vllm` to the tracker table. As of today it still requires `numpy < 2`
ref https://github.com/vllm-project/vllm/blob/cf6bbcb49324c24fc0f6f9381400c299c9c2d7ac/requirements-common.txt#L3

**Comment by rgommers** [comment]
Added, thanks @poedator. I also gave the relevant `vllm` issue a nudge - you should be able to override the pin and things will work,  since vLLM itself does not depend on NumPy and its dependencies were fixed.

**Comment by jakirkham** [comment]
Maybe CuPy support could be rewritten as 13.2.0+?

There was a 13.3.0 release, which also supported NumPy 2

A 13.4.0 release is coming up soon (in addition to the already mentioned 14)

**Comment by rootsmusic** [comment]
`[dpnp]` has been compatible since [0.16.0](https://github.com/IntelPython/dpnp/releases/tag/0.16.0).

**Comment by rgommers** [comment]
Thanks @rootsmusic, added

**Comment by Armavica** [comment]
PyTensor ([v2.28.0](https://github.com/pymc-devs/pytensor/releases/tag/rel-2.28.0)) and PyMC ([v5.21.0](https://github.com/pymc-devs/pymc/releases/tag/v5.21.0)) are now compatible

**Comment by seberg** [comment]
@Armavica thanks for the ping, added.  Happy you got this done, I expect PyTensor was one of the hardest transitions!

**Comment by jakirkham** [comment]
Looks like PySpark has done 2 RCs for 4.0.0 in the last week. So think they are close

**Comment by rootsmusic** [comment]
`pyodide` has been compatible since 0.27.0. https://github.com/pyodide/pyodide/pull/4925

**Comment by rootsmusic** [comment]
`pyodide` has been compatible since 0.27.0. https://github.com/pyodide/pyodide/pull/4925
`jupyterlite-pyodide-kernel` has been compatible since 0.5.0. https://github.com/jupyterlite/pyodide-kernel/pull/156

**Comment by agriyakhetarpal** [comment]
> `pyodide` has been compatible since 0.27.0. [pyodide/pyodide#4925](https://github.com/pyodide/pyodide/pull/4925) `jupyterlite-pyodide-kernel` has been compatible since 0.5.0. [jupyterlite/pyodide-kernel#156](https://github.com/jupyterlite/pyodide-kernel/pull/156)

Note that support for Pyodide 0.27 was also backported to `jupyterlite-pyodide-kernel` version 0.4.7 later on; this chart reflects it better: https://jupyterlite-pyodide-kernel.readthedocs.io/en/latest/#compatibility

**Comment by rgommers** [comment]
Thanks, added Pyodide and jupyterlite-pyodide-kernel. And fingers crossed for PySpark soon.

**Comment by agriyakhetarpal** [comment]
> [@agriyakhetarpal](https://github.com/agriyakhetarpal?rgh-link-date=2025-03-05T18%3A29%3A33.000Z) Do you know [which version](https://ipykernel.readthedocs.io/en/stable/changelog.html) of `ipykernel` has started supporting `numpy` 2.0?

Hi @rootsmusic, I'd say IPyKernel has anything to do with NumPy, lying in a different ecosystem altogether (Jupyter). I won't think it needs or has needed to do anything special to maintain compatibility with NumPy 2.0, unless I am mistaken somewhere. If you're referring to the [`pylab` IPython kernel mode](https://github.com/ipython/ipykernel/blob/dab3b39e3f1e0258d99b189867d8f2e2d36c976e/ipykernel/kernelapp.py#L85-L87), it should support NumPy 2.0 as long as Matplotlib is supported, which is with version 3.8.4, from the above table.

**Comment by rootsmusic** [comment]
`ngboost` 0.5.2 supports numpy 2.0. https://github.com/stanfordmlgroup/ngboost/issues/358

**Comment by rgommers** [comment]
Thanks @rootsmusic, added.

I also made an update of the tracker earlier today; it should be up to date. There's still a few stragglers, but the number of them is reducing slowly.

**Comment by dyollb** [comment]
TotalSegmentator 2.11.0 drops numpy<2.0 constraint:
https://pypi.org/project/TotalSegmentator/

**Comment by rgommers** [comment]
Okay, it's time. I will close this issue and declare victory on the ecosystem transitioning to NumPy 2.0. 🎉 

The updates on this issue have slowed down a lot over the past couple of months, and of the many packages we are tracking in this issue, only one didn't complete its support yet.

**Comment by jakirkham** [comment]
This one is now complete


| Package name                                                                     | Compatible release on PyPI? | Min compatible version                                       | Notes |
| ------------ | -------------------------------------------------------------------------------- | ---------------------- | ------------------------------------------------------------ |
| sagemaker                 |  Y                                                                                                                                                              |  2.254.0                                       | https://github.com/aws/sagemaker-python-sdk/pull/5311                                                                     |

**Comment by rgommers** [comment]
Thanks @jakirkham. I updated the table with that, given that it was the very last existing entry that still didn't have support until now. So now we're really really done:)

**Comment by mhvk** [comment]
Sorry for what may be highjacking this thread, but according to SPEC 0, we're at the end for numpy 1.x support now. Astropy right now is planning to drop support in April, but continuing to support our current 7.2 for an extra half year, to ease the transition. 

I personally would really like to drop 1.x for regular development, since numpy 2.x allows some really nice performance improvements, but there is some worry we are jumping the gun, and no-one else is planning to drop 1.x in the next half year. Is this the case? What are people planning to do (especially interested in scipy, which is one of our most used optional dependencies).

Note that there was some discussion in https://github.com/scientific-python/specs/issues/386 about making an exception to SPEC 0, but that died out. Fine to be referred to that!

**Comment by andyfaff** [comment]
> What are people planning to do (especially interested in scipy, which is one of our most used optional dependencies)

It's not clear what you mean by this? Scipy still has compat for 1.26.4, but I don't think there's a problem with binary compatibility. 
I wouldn't see any drawback with making a minimum 2.0 decision for astropy. It's a question of what your downstream project would want.

**Comment by jakevdp** [comment]
> What are people planning to do (especially interested in scipy, which is one of our most used optional dependencies).

In JAX, we follow SPEC0 for numpy/scipy dependencies.  We dropped NumPy 1.X support for the [v0.7.2 release](https://docs.jax.dev/en/latest/changelog.html#jax-0-7-2-september-16-2025) in September, which required bumping the minimum SciPy version to 1.13 four months early. It's been almost 3 months since this release, and we haven't heard any complaints that I'm aware of.

**Comment by mhvk** [comment]
@andyfaff - I meant whether for a release in the next half year, you are planning to set a minimum numpy requirement of >=2.0. In astropy, the PR removing 1.x support removes about 1000 lines (https://github.com/astropy/astropy/pull/18986), but for me the main benefit is being able to use new features (e.g., user dtypes).

And, yes, you are right, it is a question about what downstream would want, but this is not that easy to answer! My sense is that if one is stuck on numpy 1.26, it doesn't hurt to be stuck on astropy 7.2 as well, but others disagree... Anyway, really just trying to get a sense of whether people are more or less following SPEC 0 or not.

@jakevdp - thanks! Really good to hear you didn't have major problems.

**Comment by andyfaff** [comment]
Currently the next scipy release (1.17 before Christmas) will have a minimum runtime of `numpy>=1.26.4`, but the build system requires `numpy>=2.0.0`. At least that's what the pyproject.toml says, I can't remember a discussion on making 2.0 the minimum runtime. In CI we test across `main >= numpy.version.version >= 1.26.4`.
From SPEC0 it does look like we should drop 1.26.4 for 1.18.

**Comment by mhvk** [comment]
Yes, some of the people looked through larger projects and noticed they had not upped the runtime requirement to 2.0 yet, so we were curious whether people are intending to follow SPEC 0, or explicitly wait a little longer.

**Comment by lesteve** [comment]
As a data point for the scikit-learn more conservative policy (at least the current consensus see our [maintainer doc](https://scikit-learn.org/dev/developers/maintainer.html#bumping-dependencies-guideline)), it seems like we will be dropping numpy 1.x around November 2027 (i.e. in roughly 2 years). To be explicit, I don't think we are planning to have a numpy 2 specific rule, but we haven't really discussed it amongst scikit-learn maintainers ...

I don't think this is an issue per se that scikit-learn is more conservative, but if somehow you see undesired side-effects for the wider ecosystem, let us know. At the same time, to manage expectations, I won't have time to engage in long discussions about this topic [^1] until beginning January.

Explanation (see our [current rule](https://scikit-learn.org/dev/developers/maintainer.html#bumping-dependencies-guideline)):
- numpy 1.26 supports Python 3.12
- We will be requiring Python>=3.13 in November 2027. Python 3.13 was released on 2024-10-07 and our minimum Python is the one that is a bit more than 3 years old (Python has a yearly release around October)
- around November 2027, we will require Python>=3.13, numpy>=2.1.0

[^1]: for some reason I spent some effort to have scikit-learn write its own explicit rule in the context of a SPEC0 opt-in LTS variant (seems like there is some kind of movement recently, not sure what exactly ... https://github.com/scientific-python/specs/pull/389). To be honest, I don't find this topic fun or interesting so at this point I am kind of hoping to stay away from long-winded discussions around it 😅.

**Comment by mattip** [comment]
I think the discussion about dropping NumPy 1 is off-topic for this isssue, which is closed. If needed, please open a new issue or even better discuss it on one of the scientific-python forums or issue trackers.

**Comment by rgommers** [comment]
@mhvk I added a comment at https://github.com/scientific-python/specs/issues/386#issuecomment-3619804433. It's too early for dropping 1.26 for core (widely used, general-purpose) projects. AstroPy has a slightly more niche audience, so no strong opinion from me there.
